### PR TITLE
Scale input decorator label width 

### LINKED ISSUE #56848: Floating label on InputDecoration is truncated even when it could fit
b/154950540

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: SizedBox(
          width: 300,
          child: InputDecorator(
            decoration: InputDecoration(
              labelText: 'Long1 long2 long3 long4 long5 long6 text',
              border: OutlineInputBorder(
                borderSide: BorderSide(width: 1),
                borderRadius: BorderRadius.all(Radius.circular(4)),
              ),
            ),
            child: Text(''),
          ),
        ),
      ),
    );
  }
}
```

### PR DESCRIPTION
**Pull request by guidezpl**
## Description

When a text field is empty and unfocused, a label will be truncated with ellipsis if it is too long. 

When the text field is focused, a floating label will be scaled down by a factor of 0.75, but so is its width. This means the label doesn't use available space. This PR fixes this behavior by scaling the label width.

Before:
Screenshot 2020-06-08 at 22 07 46

After:
Screenshot 2020-06-08 at 22 07 14

## Related Issues

Closes #56848 

## Tests

Added a test to verify that the label width scales when focused 

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Review by justinmc — APPROVED** [review]
LGTM 👍 

I ran the branch locally and tried to break it but it seems to work in the cases I tried.

Take a look at the test that's failing, hopefully it's unrelated.

File: packages/flutter/lib/src/material/input_decorator.dart [hunk 1]
```diff
@@ -981,9 +982,11 @@ class _RenderDecoration extends RenderBox {
       + _boxSize(suffix).width
       + _boxSize(suffixIcon).width
       + contentPadding.right));
+    // Increase the available width for the label when it is scaled down.
+    final double invertedLabelScale = lerpDouble(1.00, 1/_kFinalLabelScale, decoration.floatingLabelProgress);
```
Comment (perclasson) [inline packages/flutter/lib/src/material/input_decorator.dart:986]: nit: put spaces between `1 / _kFinalLabelScale` for readability.

File: packages/flutter/test/material/input_decorator_test.dart [hunk 2]
```diff
@@ -4069,6 +4069,58 @@ void main() {
...
+    final String longStringA = String.fromCharCodes(List<int>.generate(200, (_) => 65));
+    final String longStringB = String.fromCharCodes(List<int>.generate(200, (_) => 66));
+
+    await tester.pumpWidget(
+      Center(
+        child: Container(
+          width: 100,
+          height: 100,
+          child: buildInputDecorator(
+            // isFocused: false (default)
```
Comment (perclasson) [inline packages/flutter/test/material/input_decorator_test.dart:4082]: remove this

File: packages/flutter/test/material/input_decorator_test.dart [hunk 3]
```diff
@@ -4069,6 +4069,58 @@ void main() {
...
+    );
+
+    await tester.pumpAndSettle();
+
+    expect(
+      find.text(longStringB),
+      // 133.3 is approximately 100 / 0.75 (_kFinalLabelScale)
+      paints..clipRect(rect: const Rect.fromLTWH(0, 0, 133.0, 16.0)),
+    );
+  }, skip: isBrowser);  // TODO(yjbanov): https://github.com/flutter/flutter/issues/44020);
```
Comment (perclasson) [inline packages/flutter/test/material/input_decorator_test.dart:4122]: Remove `);` at the end

**Review by perclasson — APPROVED** [review]
LGTM

File: packages/flutter/test/material/input_decorator_test.dart (hunk 2 above)
Comment (guidezpl) [reply packages/flutter/test/material/input_decorator_test.dart:4082]: It is to make it more clear and to be consistent with the other tests in this file

**Pull request merged**